src/SkyApm.Abstractions/Tracing/Segments/SegmentSpan.cs

```diff
@@ -81,7 +81,16 @@ public void AddLog(params LogEvent[] events)
 
         public void Finish()
         {
-            EndTime = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+            if (EndTime == default || EndTime < StartTime)
+            {
+                EndTime = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+            }
+        }
+
+        public void SetTimes(DateTimeOffset startTime, DateTimeOffset endTime)
```

Comment:
Why and when would you need to change manually? This is not a typical feature for SkyWalking.

Reply:
> For example, in the diagnostic information of StackExchange.Redis(SERedis), the start time and end time are determined by SERedis

I have read this, but it is not clear.

Reply:
The collection method of SERedis is to collect all commands and time objects under the current link at one time.

like this: https://github.com/StackExchange/StackExchange.Redis/blob/main/src/StackExchange.Redis/Profiling/ProfiledCommandEnumerable.cs

it contains the exact time of the beginning and end of the event, so I hope to modify the time of the SegmentSpan object.

Reply:
@liuhaoyang Your call.

Reply:
I tried to implement the `ISegmentContextFactory` interface, but still cannot modify the SegmentContext and SegmentSpan object properties.